aiida/manage/external/rmq.py

```diff
@@ -209,6 +210,10 @@ async def _continue(self, communicator, pid, nowait, tag=None):
             message = 'the class of the process could not be imported.'
             self.handle_continue_exception(node, exception, message)
             raise
+        except asyncio.CancelledError:  # pylint: disable=try-except-raise
```

Comment:
Also here: while this makes python 3.7 and 3.8 behave the same, can you briefly explain why this (re-raising rather than `handle_continue_exception` is the behavior we want?

Reply:
because `handle_continue_exception` sets the node (and thus process) as excepted, which is why all the processes currently end up as exceted when you stop the daemon in python 3.7

Reply:
Hm... I was wondering whether this was the case but then we didn't actually see the 'failed to recreate the process instance in order to continue it' message (or did we)?

In any case, if you're right, then this is an example where this change effectively results in the cancellation of the task being ignored (because we know that only the runner shutdown causes it, and we want to ignore it).

It would be good to mention in the comments the reasons for doing this, and that - eventually - the proper solution might involve handling this exception as well.

Reply:
> but then we didn't actually see the 'failed to recreate the process instance in order to continue it' message (or did we)

because it does not get to this point until you implement the other fixes I outlined

Reply:
Right. We may still want to expand the comments of this one a bit, anyhow for me it's ok to merge.

Reply:
ok cheers